mlrun/run.py

```diff
@@ -652,8 +652,11 @@ def new_function(
             runner.spec.default_handler = handler
 
     if requirements:
-        runner.with_requirements(requirements)
-    runner.verify_base_image()
+        runner.with_requirements(requirements, prepare_image_for_deploy=False)
+
+    # if a function has an image it is considered to be deployed, we prepare the image for deploy
+    # to verify that if build is required - the image will be empty
```

Comment:
this comment can be in the `prepare_image_for_deploy` instead 
```suggestion
    # if a function has a spec.image it is considered to be deployed, but because we allow the user to set spec.image  for usability purposes, we need to check whether this is a built image or it requires to be built on top
```